src/System.Net.Requests/src/System/Net/FtpWebRequest.cs

```diff
@@ -0,0 +1,2544 @@
+                                new FtpWebResponse((HttpWebResponse)webException.Response));
+                        }
+                        SetException(webException);
+                        statusCode = GetStatusCode(webException);
+                        throw webException;
+                    }
+                    // Catch added to address 
+                    catch (InvalidOperationException invalidOpException)
+                    {
+                        SetException(invalidOpException);
+                        FinishRequestStage(RequestStage.CheckForError);
+                        throw;
+                    }
+                }
+                else
+#endif
+                {
+                    RequestStage prev = FinishRequestStage(RequestStage.RequestStarted);
+                    if (prev >= RequestStage.RequestStarted)
+                    {
+                        if (prev < RequestStage.ReadReady)
+                        {
+                            lock (m_SyncObject)
+                            {
+                                if (m_RequestStage < RequestStage.ReadReady)
+                                    m_ReadAsyncResult = new LazyAsyncResult(null, null, null);
+                            }
+
+                            // GetRequeststream or BeginGetRequestStream has not finished yet?
+                            if (m_ReadAsyncResult != null)
+                                m_ReadAsyncResult.InternalWaitForCompletion();
+
+                            CheckError();
+                        }
+                    }
+                    else
+                    {
+#if FTP_FEATURE_CACHE
+                        do
+                        {
+#endif
+                            SubmitRequest(false);
+                            if (m_MethodInfo.IsUpload)
+                                FinishRequestStage(RequestStage.WriteReady);
+                            else
+                                FinishRequestStage(RequestStage.ReadReady);
+                            CheckError();
+#if FTP_FEATURE_CACHE
+                        } while (!CheckCacheRetrieveOnResponse());
+#endif
+
+                        EnsureFtpWebResponse(null);
+
+#if FTP_FEATURE_CACHE
+                        // This may update the Stream memeber on m_FtpWebResponse based on the CacheProtocol feedback.
+                        CheckCacheUpdateOnResponse();
+
+                        if (m_FtpWebResponse.IsFromCache)
+                            FinishRequestStage(RequestStage.ReleaseConnection);
+#endif
+                    }
+                }
+
+                statusCode = GetStatusCode(m_FtpWebResponse);
+#if FEATURE_FTP_TRACING
+                success = true;
+#endif
+            }
+            catch (Exception exception) {
+#if FEATURE_FTP_TRACING
+                if (FrameworkEventSource.Log.IsEnabled()) {
+                    WebException webException = exception as WebException;
```

Comment:
For CoreFX we can remove it (the ifdef, not the trace-line).
Otherwise yes, it is needed because it sets the `statusCode` that gets used later in this function in line 761.